src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/JsonIgnorePropertiesOnDataClass.kt

```diff
@@ -63,7 +77,13 @@ class JsonIgnorePropertiesOnDataClass(config: Config) : Rule(config) {
         if (kclass.isData()) {
             // Check if @JsonIgnoreProperties(ignoreUnknown = true) annotation exists
             val hasJsonIgnoreProperties = kclass.annotationEntries.any { annotation ->
-                annotation.shortName?.asString() == JSON_IGNORE_ANNOTATION &&
+                // Get fully qualified import paths
+                val fullyQualifiedImportNames = kclass.containingKtFile.importDirectives.mapNotNull { importDirective ->
```

Comment:
what happens if someone does a `import com.fasterxml.jackson.annotation.*`?

are you sure there's nothing in `annotation` that can help?

Reply:
checking that use case

Reply:
I will have to check for other functions in the library then, because I am not able to find anyway to match the import when wildcard imports are present. I could do a startsWith(com.fasterxml.jackson.) to check if it is the same library or not.

Reply:
I have tested with startsWith and it is working, please check once. I am not able to find any other method which checks the import of the annotation itself. @thewisenerd would you know of any way?

Reply:
i've refactored it to use proper FQ names, hopefully not breaking any functionality

https://github.com/udaan-com/detekt-sidekt/pull/20/commits/628cf5345473ad428233f2abe1d63a463d6c1347

Reply:
also added a test for wrong import